File: ios/SentryTags.swift
import Foundation

/// Tag keys we set on Sentry events. Mirrors `SentryTags.kt` and
/// `src/sentry-tags.ts` so `proc` / `layer` etc. are spelled
/// identically across all three layers.
enum SentryTags {
    static let proc = "proc"
    static let layer = "layer"
    static let phase = "comapeo.phase"
    static let state = "comapeo.state"
    static let source = "source"
    static let timeout = "timeout"
    static let timeoutMs = "timeoutMs"

    // proc values — iOS is single-process, so always "main"
    static let procMain = "main"

    // layer values
    static let layerRn = "rn"
    static let layerNative = "native"
    static let layerNode = "node"
}

File: src/sentry-tags.ts
/**
 * Tag keys we set on Sentry events. Centralised so a typo can't
 * silently route an event to the wrong dashboard column.
 *
 * `proc` reflects the actual OS process: iOS is always `main`;
 * Android is `main` for RN/native code in the host UI process and
 * `fgs` for code in the `:ComapeoCore` foreground-service process
 * (Kotlin FGS code AND the embedded nodejs-mobile that runs there).
 */
export const SentryTags = {
  proc: "proc",
  layer: "layer",
  phase: "comapeo.phase",
  state: "comapeo.state",
  source: "source",
  timeout: "timeout",

  procMain: "main",
  procFgs: "fgs",

  layerRn: "rn",
  layerNative: "native",
  layerNode: "node",
} as const;

/**
 * Breadcrumb category names. Single source of truth for the
 * dot-separated category strings so a typo can't silently route
 * crumbs to the wrong dashboard filter.
 */
export const SentryCategories = {
  /** State-machine transitions (STOPPED → STARTING → STARTED …). */
  state: "comapeo.state",
  /** Control-socket parse failures (the `messageerror` event). */
  control: "comapeo.control",
} as const;
